components/core/JsInterop/modules/components/uploadHelper.ts

```diff
@@ -57,7 +57,7 @@ export class uploadHelper {
     return url;
   }
 
-  static uploadFile(element, index, data, headers, fileId, url, name, instance, percentMethod, successMethod, errorMethod) {
+  static uploadFile(element, index, data, headers, fileId, url, name, instance, percentMethod, successMethod, errorMethod, method: string) {
```

Comment:
I think it would be better if instead of `string` a string literal was used.

Reply:
Do I understand correctly that you're suggesting it be `method: "GET" | "HEAD" | "POST" | "PUT" | "DELETE" | "CONNECT" | "OPTIONS" | "TRACE" | "PATCH"` ?

Note that this would be case sensitive, and although the HTTP spec says the methods should be upper case, the existing implementation defaults to a lower case `post`.

I also worry that this would be overly restrictive (if a new http method were to be introduced), and inconsistent with most other typescript definitions for the XHR open method (Examples: http://definitelytyped.org/docs/jquery--jquery/interfaces/jqueryxhr.html#open , https://github.com/hmoog/xmlhttprequest-ts/blob/master/src/XMLHttpRequest.ts)

Reply:
I get your point. I think adding such string literal or using an enum would be self documenting, but I will not insist on it.